bundles/org.openhab.binding.sbus/src/main/java/org/openhab/binding/sbus/handler/AbstractSbusHandler.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sbus.handler;
+
+import static org.openhab.binding.sbus.BindingConstants.BINDING_ID;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.sbus.internal.SbusBridgeHandler;
+import org.openhab.binding.sbus.internal.config.SbusDeviceConfig;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ro.ciprianpascu.sbus.facade.SbusAdapter;
+
+/**
+ * The {@link AbstractSbusHandler} is the base class for all SBUS device handlers.
+ * It provides common functionality for device initialization, channel management, and polling.
+ *
+ * @author Ciprian Pascu - Initial contribution
+ */
+@NonNullByDefault
+public abstract class AbstractSbusHandler extends BaseThingHandler {
+
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    protected @Nullable SbusAdapter sbusAdapter;
+    protected @Nullable ScheduledFuture<?> pollingJob;
+
+    public AbstractSbusHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public final void initialize() {
+        logger.debug("Initializing SBUS handler for thing {}", getThing().getUID());
+
+        initializeChannels();
+
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+            return;
+        }
+
+        SbusBridgeHandler bridgeHandler = (SbusBridgeHandler) bridge.getHandler();
+        if (bridgeHandler == null || bridgeHandler.getThing().getStatus() != ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge is not online");
+            return;
+        }
+
+        try {
+            sbusAdapter = bridgeHandler.getSbusConnection();
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
As the `getSbusConnection` is Nullable, i don't think it is right to set the thing to online, as you are not yet sure there is an actual connection. Maybe better to move this state update to the polling job. Every time a poll is succesfull / data flows it can be set to online.

Edit: looking at the child handlers, the polling seems to do exactly this, so i would say this updateStatus can be removed here.